label_studio/users/forms.py

```diff
@@ -87,7 +89,16 @@ def save(self):
         allow_newsletters = None
         if 'allow_newsletters' in cleaned:
             allow_newsletters = cleaned['allow_newsletters']
-        user = User.objects.create_user(email, password, allow_newsletters=allow_newsletters)
+        try:
+            user = User.objects.create_user(email, password, allow_newsletters=allow_newsletters)
+        except IntegrityError:   # user already exists (often this means a soft deleted user is being reactivated)
+            user = fast_first(User.with_deleted.filter(email=email))
+            if user and user.is_deleted:
```

Comment:
`invite_user_to_organization` exists in LSE and is responsible for actually saving the user model if it was a deleted user.  Its unfortunate the feature must be spread over the two repositories but its a consequence of how user creation is also spread between LS and LSE.  Regardless, in LS this will never result in a user being created as the user is not subsequently saved.